library/core/src/ops/deref.rs

```diff
@@ -309,3 +309,25 @@ impl<T: ?Sized> Receiver for &T {}
 
 #[unstable(feature = "receiver_trait", issue = "none")]
 impl<T: ?Sized> Receiver for &mut T {}
+
+#[lang = "stable_deref"]
+#[unstable(feature = "stable_deref_trait", issue = "123430")]
+/// # Safety
+///
+/// Any two calls to `deref` must return the same value at the same address unless
```

Comment:
"same value at the same address" here is pretty vague, and it's unclear to me without reading the RFC thread what exactly that entails and how it compares to the stable deref trait crate. notably, the preconditions on the crate are currently **not** satisfied by Box because of the special strict aliasing rules imposed by it.

Reply:
The meaning of "same value" is that the concrete type must not change. I copied the explanation in below. Was it unclear?

> Here, "same value" means that if `deref` returns a trait object, then the actual type behind that trait object must not change. Additionally, when you unsize coerce from `Self` to `Unsized`, then if you call `deref` on `Unsized` and get a trait object, then the underlying type of that trait object must be `<Self as Deref>::Target`.
>
> Analogous requirements apply to other unsized types. E.g., if `deref` returns `[T]`, then the length must not change. In other words, the underlying type must not change from `[T; N]` to `[T; M]`.